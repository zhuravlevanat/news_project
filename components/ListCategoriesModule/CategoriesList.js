import {connect} from "react-redux";
import NewsLink from "../NewsPageModule/NewsLink";
import { getCategories } from "../../redux/Categories/selector";
import React, {Fragment} from "react";
import List from '@material-ui/core/List';
import {getLoadingState} from "../../redux/News/selectors";
import Spinner from "../helpers/Spinner";

const mapStateToProps = (state) => {
    return {
        categories: getCategories(state),
        loading: getLoadingState(state),
    }
}

const CategoriesList = ({categories, loading}) => {
    return (
        <Fragment>
            {loading
                ? <Spinner />
                : (
                    <List component="ul" aria-label="news categories">
                        {categories.map((title, index) => <NewsLink title={title} key={index}/>)}
                    </List>
                )
            }

        </Fragment>
    )
};

export default connect(mapStateToProps)(CategoriesList);