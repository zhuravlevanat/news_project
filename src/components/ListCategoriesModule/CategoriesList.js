import NewsLink from "../helpers/NewsLink";
import React from "react";
import List from '@material-ui/core/List';
import Spinner from "../helpers/Spinner";

const CategoriesList = ({ categories, loading }) => {
    return (
        <>
            {loading
                ? <Spinner />
                : (
                    <List component="ul" aria-label="news categories">
                        {categories.map((title, index) => <NewsLink title={title} key={index}/>)}
                    </List>
                )
            }
        </>
    )
};

export default CategoriesList;
