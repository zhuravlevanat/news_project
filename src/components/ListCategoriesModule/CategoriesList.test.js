import React from "react";
import { shallow } from 'enzyme';
import CategoriesList from "./CategoriesList";

describe('CategoriesList', () => {
    const props = {
        categories: [],
        loading: false,
    };
    describe('CategoriesList initial', () => {
        const list = shallow(<CategoriesList {...props} />);
        it('renders properly', () => {
            expect(list).toMatchSnapshot();
        });

        it('render initial', () => {
            expect(list.find('Spinner')).toHaveLength(0);
            expect(list.find(`WithStyles(ForwardRef(List))`)).toHaveLength(1);
        });
    });
    describe('CategoriesList loading', () => {
        const nextProps = {
            ...props,
            loading: true,
        };
        const list = shallow(<CategoriesList {...nextProps} />);

        it('renders properly', () => {
            expect(list).toMatchSnapshot();
        });

        it('render loading', () => {
            expect(list.find('Spinner')).toHaveLength(1);
            expect(list.find(`WithStyles(ForwardRef(List))`)).toHaveLength(0);
        });
    });
    describe('NewsList with loading', () => {
        const nextProps = {
            ...props,
            categories: ['food', 'drinks'],
            loading: false,
        };
        const list = shallow(<CategoriesList {...nextProps} />);
        console.log(list.debug());
        it('render loading', () => {
            expect(list.find('ListItemLink')).toHaveLength(nextProps.categories.length);
        });
    });
});
