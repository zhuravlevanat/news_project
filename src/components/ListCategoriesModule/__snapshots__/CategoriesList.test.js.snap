// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList CategoriesList initial renders properly 1`] = `ShallowWrapper {}`;

exports[`CategoriesList CategoriesList loading renders properly 1`] = `ShallowWrapper {}`;

exports[`CategoriesList News list initial renders properly 1`] = `ShallowWrapper {}`;

exports[`CategoriesList NewsList with loading renders properly 1`] = `ShallowWrapper {}`;
